#!/usr/bin/env python3
PACKAGE = "spot_fetch"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("hue_min", double_t, 0, "Min hue (0-180\u00b0)", 32.12121212121213, 0, 180)
gen.add("hue_max", double_t, 0, "Max hue (0-180\u00b0)", 77, 0, 180)
gen.add("sat_min", double_t, 0, "Min saturation (0-255)", 30.0, 0, 255)
gen.add("sat_max", double_t, 0, "Max saturation (0-255)", 130.0, 0, 255)
gen.add("val_min", double_t, 0, "Min value (0-255)", 108.0, 0, 255)
gen.add("val_max", double_t, 0, "Max value (0-255)", 255.0, 0, 255)
gen.add("contours_min_area", double_t, 0, "contours_min_area", 5.0, 0, 100)
gen.add("contours_min_perimeter", double_t, 0, "contours_min_perimeter", 0, 0, 100)
gen.add("contours_min_width", double_t, 0, "contours_min_width", 0, 0, 100)
gen.add("contours_max_width", double_t, 0, "contours_max_width", 1000, 0, 1000)
gen.add("contours_min_height", double_t, 0, "contours_min_height", 0, 0, 100)
gen.add("contours_max_height", double_t, 0, "contours_max_height", 1000, 0, 1000)
gen.add("contours_min_solidity", double_t, 0, "contours_min_solidity", 0.0, 0, 100)
gen.add("contours_max_solidity", double_t, 0, "contours_max_solidity", 100, 0, 100)
gen.add("contours_max_vertices", double_t, 0, "contours_max_vertices", 1000000, 0, 1000000)
gen.add("contours_min_vertices", double_t, 0, "contours_min_vertices", 0, 0, 100)
gen.add("contours_min_ratio", double_t, 0, "contours_min_ratio", 0.75, 0, 100)
gen.add("contours_max_ratio", double_t, 0, "contours_max_ratio", 1000, 0, 1000)

exit(gen.generate(PACKAGE, "fetch", "TennisBall"))
